### PR TITLE
Create CLI commands for AWS auth manager to create Amazon Verified Permissions related resources

### PR DESCRIPTION
**Pull request by vincbeck**
AWS auth manager uses Amazon Verified Permissions service as authorizer. This PR creates two CLI commands in order to help creating the resources needed for the AWS auth manager to work:
- `init_avp`. Initialize the resources needed in Amazon Verified Permissions
- `update_avp_schema`. Update the Amazon Verified Permissions policy store schema to the latest version

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/auth_manager/cli/avp_commands.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    """Initialize Amazon Verified Permissions resources."""
+    client = _get_client()
+
+    # Create the policy store if needed
+    policy_store_id, is_new_policy_store = _create_policy_store(client, args)
+
+    if not is_new_policy_store:
+        print(
+            f"Since an existing policy store with description '{args.policy_store_description}' has been found in Amazon Verified Permissions, "
+            "the CLI makes no changes to this policy store for security reasons. "
```
Comment (o-nikolas) [inline airflow/providers/amazon/aws/auth_manager/cli/avp_commands.py]:
```suggestion
            "the CLI made no changes to this policy store for security reasons. "
```

File: airflow/providers/amazon/aws/auth_manager/cli/avp_commands.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    if len(existing_policy_stores) > 0:
+        print(
+            f"There is already a policy store with description '{args.policy_store_description}' in Amazon Verified Permissions: '{existing_policy_stores[0]['policyStoreId']}'."
+        )
+        return existing_policy_stores[0]["policyStoreId"], False
+    else:
+        print(f"No policy store with description '{args.policy_store_description}' found, creating one.")
+        if args.dry_run:
+            print("Dry run, not creating the policy store.")
```
Comment (o-nikolas) [inline airflow/providers/amazon/aws/auth_manager/cli/avp_commands.py]: maybe include the inputs you would have provided to the create_policy_store api in the dry run log message (same for the other methods as well)? That might make the dry run command more useful.

File: tests/providers/amazon/aws/auth_manager/cli/test_avp_commands.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                "--policy-store-description",
+                policy_store_description,
+            ]
+            if dry_run:
+                params.append("--dry-run")
+            if verbose:
+                params.append("--verbose")
+            init_avp(self.arg_parser.parse_args(params))
+
+        if not dry_run:
```
Comment (o-nikolas) [inline tests/providers/amazon/aws/auth_manager/cli/test_avp_commands.py]: Nothing is being asserted for the dry_run case. So as long as there is no uncaught exception the dry_run == true case is an automatic pass?

I think it's worth asserting the boto3 apis are not called/called zero times in that case.
Same with the schema test

File: airflow/providers/amazon/aws/auth_manager/cli/avp_commands.py (hunk 2 above)
Comment (vincbeck) [reply airflow/providers/amazon/aws/auth_manager/cli/avp_commands.py]: Fair point

File: tests/providers/amazon/aws/auth_manager/cli/test_avp_commands.py (hunk 3 above)
Comment (vincbeck) [reply tests/providers/amazon/aws/auth_manager/cli/test_avp_commands.py]: Nice catch! Thanks

**Review by o-nikolas — APPROVED** [review]
Code lgtm!

**Comment by potiuk** [comment]
For whatever reason this one fails Airflow compatibility checks in `canary` build. I will take a look why it did not fail one here https://github.com/apache/airflow/actions/runs/7539705871/job/20523381688

**Comment by potiuk** [comment]
But it would be great if you could take a look @vincbeck @o-nikolas - maybe it will be obvious to you

**Comment by o-nikolas** [comment]
> But it would be great if you could take a look @vincbeck @o-nikolas - maybe it will be obvious to you

It's a very weird import error. That module certainly exists at that path and has existed for some time now. Very odd...

**Comment by vincbeck** [comment]
Following up on this one, is it resolved or we need to do something about it? @Taragolis

**Comment by potiuk** [comment]
https://github.com/apache/airflow/pull/36824 should fix it.

**Comment by potiuk** [comment]
> It's a very weird import error. That module certainly exists at that path and has existed for some time now. Very odd...

Nope. It's been added in 2.7.0.

**Comment by potiuk** [comment]
And I even know why it did not fail in PR but started to fail only in SOME PRs and main, "canary" build.

The problem is that in PR you do not run "full checks"  - only a subset of the tests. And we only build and install modified providers to save time. The problem in this case is that when we do not build all providers - only few - this check only INSTALLS providers but does not VERIFY them - because verification might fail if we only install few providers, because of some dynamic cross-importing between providers.

That's why this check only fails in those PRs that - for whatever reason - build and install and VERIFY all providers. 

However, I think this speed-up is not needed any more. It gave a lot savings when building providers took way more time - and I optimized it a lot since (like 10x) and we can easily build all providers now for all PRs as it takes less than 1 minute now (it used to takes 12 minutes or so).

**Comment by potiuk** [comment]
Fix for future similar cases is coming.

**Pull request merged**